src/box/memtx_allocator.h

```diff
@@ -286,107 +325,150 @@ class MemtxAllocator {
 	}
 
 	/**
-	 * Allocate a tuple of the given size.
+	 * True if a block is visible from at least one read view.
+	 * In that case it cannot be modified.
 	 */
-	static struct tuple *alloc_tuple(size_t size)
+	static bool
+	in_read_view(const struct memtx_block *block, bool is_temporary = false)
```

Comment:
By the way, if the block is in read view, you have to reallocate it and only then modify it in place. Wouldn't it be handy to have a `static struct memtx_block *realloc(const struct memtx_block *)` method as well?

Reply:
"Reallocate" sounds confusing, because the real `realloc` frees the old block, but here we need both copies. Let's do the copy on the user side.

Reply:
We could call this method `copy` then. Otherwise, we would need to implement copying in each module that uses `memtx_block`.

Anyway, let's think about it later, when we will actually use it somewhere.